SpatialGDK/Source/SpatialGDKServices/Private/LocalReceptionistProxyServerManager.cpp

```diff
@@ -0,0 +1,265 @@
+
+#include "Internationalization/Regex.h"
+#include "Misc/FileHelper.h"
+#include "Serialization/JsonReader.h"
+#include "Serialization/JsonSerializer.h"
+#include "Serialization/JsonWriter.h"
+#include "Sockets.h"
+#include "SocketSubsystem.h"
+#include "UObject/CoreNet.h"
+
+#include "SpatialCommandUtils.h"
+#include "SpatialGDKServicesConstants.h"
+#include "..\Public\LocalReceptionistProxyServerManager.h"
+
+DEFINE_LOG_CATEGORY(LogLocalReceptionistProxyServerManager);
+
+#define LOCTEXT_NAMESPACE "FLocalReceptionistProxyServerManager"
+
+namespace
+{
+	static const FString ProxyInfoFilePath = FPaths::Combine(SpatialGDKServicesConstants::ProxyFileDirectory, SpatialGDKServicesConstants::ProxyInfoFilename);
+}
+
+FLocalReceptionistProxyServerManager::FLocalReceptionistProxyServerManager()
+: RunningCloudDeploymentName(TEXT(""))
+{
+}
+
+bool FLocalReceptionistProxyServerManager::GetProcessName(const FString& PID, FString& OutProcessName)
+{
+	bool bSuccess = false;
+	OutProcessName = TEXT("");
+	const FString TaskListCmd = TEXT("tasklist");
+
+	// Get the task list line for the process with Pid 
+	const FString TaskListArgs = FString::Printf(TEXT(" /fi \"PID eq %s\" /nh /fo:csv"), *PID);
+	FString TaskListResult;
+	int32 ExitCode;
+	FString StdErr;
+	bSuccess = FPlatformProcess::ExecProcess(*TaskListCmd, *TaskListArgs, &ExitCode, &TaskListResult, &StdErr);
+	if (ExitCode == 0 && bSuccess)
+	{
+		FRegexPattern ProcessNamePattern(TEXT("\"(.+?)\""));
+		FRegexMatcher ProcessNameMatcher(ProcessNamePattern, TaskListResult);
+		if (ProcessNameMatcher.FindNext())
+		{
+			OutProcessName = ProcessNameMatcher.GetCaptureGroup(1 /* Get the Name of the process, which is the first group. */);
+
+			return true;
+		}
+	}
+
+	UE_LOG(LogLocalReceptionistProxyServerManager, Warning, TEXT("Failed to get the name of the process that is blocking the required port."));
+	
+	return false;
+}
+
+bool FLocalReceptionistProxyServerManager::CheckIfPortIsBound(int32 Port, FString& OutPID, FString& OutLogMsg)
+{
+	FString State;
+	FString ProcessName;
+	bool bSuccess = SpatialCommandUtils::GetProcessInfoFromPort(Port, OutPID, State);
+	if (bSuccess)
+	{
+		if (State.Contains("LISTENING"))
+		{
+			if (GetProcessName(OutPID, ProcessName))
+			{
+				OutLogMsg = TEXT("%s process with PID: %s"), ProcessName, OutPID;
+				return true;
+			}
+
+			OutLogMsg = TEXT("Unknown process with PID: %s."), *OutPID;
+
+			return true;
+		}
+	}
+	OutLogMsg = TEXT("No Process is blocking the required port.");
+
+	return false;
+}
+
+bool FLocalReceptionistProxyServerManager::LocalReceptionistProxyServerPreRunChecks(int32 ReceptionistPort)
+{
+	FString OutLogMessage;
+	FString PID;
+
+	// Check if any process is blocking the receptionist port
+	if (CheckIfPortIsBound(ReceptionistPort, PID, OutLogMessage))
+	{
+		// Try killing the process that blocks the receptionist port if the process blocking the port is a previously running proxy.
+		if (PID == ParsePid())
```

Comment:
Could you rename `ParsePid` so it's more descriptive? e.g. `GetPreviousReceptionistProxyPID`